packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -178,16 +184,30 @@ class DependencyExtractionWebpackPlugin {
 					continue;
 				}
 
-				const assetFilename = buildFilename.replace(
-					/\.js$/i,
-					'.asset.' + ( outputFormat === 'php' ? 'php' : 'json' )
-				);
+				let assetFilename;
+
+				if ( outputFilename ) {
+					assetFilename = compilation.getPath( outputFilename, {
```

Comment:
Should we enforce a matching `outputFormat` based on the extension used in the file? At least when it ends with `.php` or `.json`. We could also leave it as is - the developer using the plugin still would have to pass two options to set everything correctly when using non-default output format.